code/obj/machinery/runetrap.dm

```diff
@@ -0,0 +1,226 @@
+//////////////////////////
+// Light revealed wraith trap
+//////////////////////////
+/obj/machinery/wraith/runetrap
+	name = "Rune trap"
+	desc = "A strange ominous circle. You should likely tip-toe around this one."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "rune_trap"
+	density = 0
+	anchored = 1
+	var/visible = FALSE
+	var/armed = FALSE
+	var/mob/wraith/wraith_trickster/master = null
+
+	New(var/turf/T, var/mob/wraith/wraith_trickster/W = null)
+		..()
+		master = W
+		SPAWN(5 SECONDS)
+			var/turf/local_turf = get_turf(src)
+			if (local_turf.RL_GetBrightness() < 0.3)
+				APPLY_ATOM_PROPERTY(src, PROP_MOB_INVISIBILITY, src, INVIS_CLOAK)
+				animate(src, alpha=120, time = 1 SECONDS)
+			src.armed = TRUE
+
+	process()
+		..()
+		if (!src.armed)	//Wait until we are armed.
+			return 1
+		var/turf/local_turf = get_turf(src)
+		if (local_turf.RL_GetBrightness() < 0.3 && src.visible)
+			APPLY_ATOM_PROPERTY(src, PROP_MOB_INVISIBILITY, src, INVIS_CLOAK)
+			src.visible = FALSE
+			animate(src, alpha=120, time = 1 SECONDS)
+		else if (local_turf.RL_GetBrightness() >= 0.2 && !src.visible)
+			REMOVE_ATOM_PROPERTY(src, PROP_MOB_INVISIBILITY, src)
+			src.visible = TRUE
+			animate(src, alpha=255, time = 1 SECONDS)
+			src.visible_message("<span class='alert>[src] is revealed!")
+
+	attackby(obj/item/P, mob/living/user)
+		playsound(src, "sound/impact_sounds/Crystal_Shatter_1.ogg", 80)
+		src.visible_message("<span class='notice'>The trap is destroyed!</span>")
+		qdel(src)
+
+	disposing()
+		. = ..()
+		if (master != null)
+			master.traps_laid--
+
+/obj/machinery/wraith/runetrap/madness
+	var/amount_to_inject = 8
+
+	Crossed(atom/movable/A)
+		..()
+		if(!src.armed)
+			return 1
+		if(ismob(A) && !istype(A, /mob/living/critter/wraith/trickster_puppet) && !istype(A, /mob/wraith))
```

Comment:
```suggestion
		if(ismob(A) && !isintangible(A) && !istype(A, /mob/living/critter/wraith/trickster_puppet) && !istype(A, /mob/wraith))
```
not 100% sure Crossed triggers for intangible mobs, but if it does we need this, also consider putting this check in the parent and calling `src.triggered()` or something so you're not repeating it in every subtype